iwyu_output.cc

```diff
@@ -613,6 +627,28 @@ void IwyuFileInfo::ReportForwardDeclareUse(SourceLocation use_loc,
   LogSymbolUse("Marked fwd-decl use of decl", symbol_uses_.back());
 }
 
+void IwyuFileInfo::ReportUsingDeclUse(SourceLocation use_loc,
+                                      const UsingDecl* using_decl,
+                                      bool in_cxx_method_body,
+                                      const char* comment) {  
+  // If accessing a symbol through a using decl in the same file that contains
+  // the using decl, we must mark the using decl as referenced. At the end of
+  // traversing the AST, we check to see if a using decl is unreferenced and
+  // add a full use of one of its shadow decls so that the source file
+  // continues to compile.
+  map<const UsingDecl*, bool>::iterator using_decl_status = 
+    using_decl_referenced_.find(using_decl);
+
+  if (using_decl_status != using_decl_referenced_.end()) {
+    using_decl_status->second = true;
+  }
+
```

Comment:
Not found is OK because here, we are looking to see if a using decl is referenced in the same file it is declared in. Only then do we care to mark the using decl as referenced. This is strictly to cover the case where a file locally declares a using decl but doesn't actually use it, so we need to force the include of a symbol that satisfies the the using decl;  

ie: if we have;

```cpp
#include <vector>
using std::vector;
```
But then never actually use a vector, we will try to remove the vector #include and break compilation.